image/cli/mascli/functions/gitops_deprovision_cos

```diff
@@ -185,19 +185,180 @@ function gitops_deprovision_cos() {
   sm_login
 
   export COS_ACTION="deprovision"
-  export COS_SECRET=${ACCOUNT_ID}${SECRETS_KEY_SEPERATOR}${CLUSTER_ID}${SECRETS_KEY_SEPERATOR}${MAS_INSTANCE_ID}${SECRETS_KEY_SEPERATOR}cos
   export MAS_CONFIG_DIR=$TEMP_DIR
   export MAS_INSTANCE_ID=${MAS_INSTANCE_ID:-CLUSTER_ID}
-  export ROLE_NAME=cos && ansible-playbook ibm.mas_devops.run_role
 
-  rc=$?
-  [ $rc -ne 0 ] && exit $rc
+  if [[ "$COS_TYPE" == "ibm" ]]; then
+
+    export COS_SECRET=${ACCOUNT_ID}${SECRETS_KEY_SEPERATOR}${CLUSTER_ID}${SECRETS_KEY_SEPERATOR}${MAS_INSTANCE_ID}${SECRETS_KEY_SEPERATOR}cos
+    export ROLE_NAME=cos && ansible-playbook ibm.mas_devops.run_role
+
+    rc=$?
+    [ $rc -ne 0 ] && exit $rc
+
+    echo -e "Deleting secret  $COS_SECRET"
+    sm_delete_secret $COS_SECRET
+
+  elif [[ "$COS_TYPE" == "aws" ]]; then
+
+    curl https://releases.hashicorp.com/terraform/1.9.2/terraform_1.9.2_linux_amd64.zip --output $TEMP_DIR/terraform_1.9.2_linux_amd64.zip
```

Comment:
Rather than install terraform here we should install it as part of the cli-base image here https://github.com/ibm-mas/cli-base/tree/stable/image/cli-base/install so that it is present all the time.

Reply:
Terence stated it is something we shouldn't ship to customers.  Does that effect how it should be installed?  Would licensing effect the decision whether to put it in the cli-base or not?